[yamui] More reliable displaying of content over devices. Contributes: JB#14745

philippedeswert opened the pull request.

philippedeswert committed:
[yamui] clean-up and improvements

Make yamui build cleanly, and improve the linking

Signed-off-by: Philippe De Swert <philippe.deswert@jollamobile.com>

philippedeswert committed:
[rpm] Fix some rpmlint warnings

 * strip binary on install
 * do not explicitly depend on libpng but let rpm sort it out
 * supply url

Signed-off-by: Philippe De Swert <philippe.deswert@jollamobile.com>

thp commented on minui/graphics_fbdev.c:
```diff
@@ -179,6 +180,20 @@ static gr_surface fbdev_init(minui_backend* backend) {
 static gr_surface fbdev_flip(minui_backend* backend __attribute__((unused))) {
+
+#if defined(RECOVERY_BGRA)
+        // In case of BGRA, do some byte swapping
```
Would it be enough to just flip the R and B component inside `gr_color()` (and possibly other places that deal with color)? This way, BGRA support would come at the same cost as RGBA support, whereas with this approach, we have to scan through the whole framebuffer to swap bytes.

philippedeswert replied:
Unfortunately it only seems to work with the text, not with the images. Anyway it is not needed in normal conditions, so they are there for reference if we need them.

thp replied:
Oh yeah, didn't think of images. Makes sense to have it there for reference.

philippedeswert committed:
[minui] Add extra fbdev ioctl and more potentially needed transforms

Sometimes the fb needs to be told what we expect it to be to start
showing things. Also in some cases it might be needed to add at least
an alpha channel to get things to show. (compile time option)


Signed-off-by: Philippe De Swert <philippe.deswert@jollamobile.com>

thp commented:
LGTM.

philippedeswert committed:
[fbdev] Allow FBIOPUT_VSCREENINFO ioctl to fail

Even if the ioctl fails, in many cases things might still work.
As in some cases it is required to make display work, we let it fail
without quitting.

Signed-off-by: Philippe De Swert <philippe.deswert@jollamobile.com>

philippedeswert committed:
[clean-up] Remove dead test code

The test code does not compile and calls functions that do not exist,
so let's just remove it.

Signed-off-by: Philippe De Swert <philippe.deswert@jollamobile.com>

philippedeswert committed:
[ioctl] Save the original fb info

Seems that we need to make sure the original fb info is still there
if the FBIOPUT_VSCREENINFO ioctl fails.


Signed-off-by: Philippe De Swert <philippe.deswert@jollamobile.com>
